### PR TITLE
fix: tile minimize to tray apps

### LINKED ISSUE #1009: Sleek App can't be controlled with tiling wm commands with minimize to tray enabled
**(1) Issue/Bug Description:**
The [Sleek app](https://www.flathub.org/apps/details/com.github.ransome1.sleek) can't be controlled by tiling wm commands if minimize to tray bar is set to true in the settings. I can't change its location using the keyboard.

**(2) Steps to reproduce (if you know):**
install sleek.
set the minimize to tray bar option to true in the settings
Try to use commands to control the window position (super + enter, then arrow keys)

**(3) Expected behavior:**
Window tiles and moves about the various monitors

**(4) Distribution:**
 ```
NAME="Pop!_OS"
VERSION="20.10"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 20.10"
VERSION_ID="20.10"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=groovy
UBUNTU_CODENAME=groovy
LOGO=distributor-logo-pop-os
```

**(5) Gnome Shell version:**
`3.38.3-2ubuntu0.20.10.2pop1~1614116211~20.10~0176ca0`

**(6) Pop Shell version:**
```
pop-shell:
  Installed: 1.1.0~1624036583~20.10~93634d6
  Candidate: 1.1.0~1624036583~20.10~93634d6
  Version table:
 *** 1.1.0~1624036583~20.10~93634d6 1001
       1001 http://ppa.launchpad.net/system76/pop/ubuntu groovy/main amd64 Packages
       1001 http://ppa.launchpad.net/system76/pop/ubuntu groovy/main i386 Packages
        100 /var/lib/dpkg/status
```

**(7) Where was Pop Shell installed from:**
pop-os

**(8) Monitor Setup:**
built in 1080p display, with or without external monitors

**(9) Other Installed/Enabled Extensions:**
pop-os default extensions

**(10) Other Notes:**
This only happens when the minimize to tray bar is set to true, which I can live without. So it's not a high priority bug, but I thought I'd mention it, because perhaps something important can be learned from this bug. FYI Sleek is an electron app.

p.s. edited to correct my earlier faulty understanding of the cause of the bug

### PR DESCRIPTION
**Pull request by jmmaranan**
Fixes #1009

- Installed Sleek via Flatpak
- Update Settings to Minimize to Tray

### CODE REVIEW

**Review by bflanagin — APPROVED** [review]
Sleek now moves as expected using pop-shell.

File: src/window.ts [hunk 1]
```diff
@@ -258,9 +258,8 @@ export class ShellWindow {
                 wm_class = this.name(ext)
             }
 
-            return !this.meta.is_skip_taskbar()
```
Comment (jackpot51) [inline src/window.ts:261]: @mmstick do you know what this line was supposed to do?

**Comment by bflanagin** [comment]
Adding a not about Sleek in general that isn't connected to this PR, but worth noting. 
With minimize to tray bar is enabled, sleek no longer shows as a window in any of the usual places. (Workspace spread, Super+Tab,Workspace preview) This happens before and after the PR so it isn't a regression.

**Comment by jacobgkau** [comment]
Please hold on our approval for this, we need to do regression testing since this is merging into `master`.

File: src/window.ts (hunk 1 above)
Comment (mmstick) [reply src/window.ts:261]: That I don't recall.

**Comment by bflanagin** [comment]
After continued testing , The issues caused by having "Minimized to tray bar" enabled in Sleek have a broader reach than being able to move the window via the keyboard. All of which can be avoided (including the keyboard control issue) by not having "Minimize to tray bar" enabled. These visual issues are present before and after the PR. So at the moment the fix, though a step in the right direction, doesn't add any benefits that could be achieved by staying with the defaults within the application in question.  

 I've included a video of some of the visual issues. 

https://user-images.githubusercontent.com/10382037/123134441-01d06f80-d40e-11eb-88cb-e685a9579415.mp4

I suggest moving this PR to draft and looking into the other visual issues.

**Comment by jmmaranan** [comment]
Probably need to research if gnome shell does not track skip task bar windows on Overview/Workspace on non pop-shell present like stock Ubuntu.

I assumed the user spends time on tiling to rearrange windows vs doing them on overview that's why I added the PR. But maybe a wrong assumption.

**Comment by jmmaranan** [comment]
I'll also see how i3 handles this UX minus an overview

**Comment by bflanagin** [comment]
@jmmaranan, don't get me wrong, its an improvement, and what your PR says it does, it does.  

I over stepped my bounds when reviewing the PR, and will create issues for each of the other idiosyncrasies when that option is enabled under Pop Shell.

**Comment by jmmaranan** [comment]
@bflanagin - all good. I thought it would be interesting issue to solve.

**Review by bflanagin — APPROVED** [review]
PR fixes the issue it reports to. 

All regression tests pass on the follow:
[shell-regression-checklist-1021-focal.txt](https://github.com/pop-os/shell/files/6704825/shell-regression-checklist-1021-focal.txt)
[shell-regression-checklist-1021-groovy.txt](https://github.com/pop-os/shell/files/6704826/shell-regression-checklist-1021-groovy.txt)
[shell-regression-checklist-1021-hirsute.txt](https://github.com/pop-os/shell/files/6704827/shell-regression-checklist-1021-hirsute.txt)

**Comment by mmstick** [comment]
For what it's worth, it's acceptable for us to override GNOME Shell behaviors to achieve our desired goals.

**Pull request merged**